app/consumer.py

```diff
@@ -0,0 +1,119 @@
+        super().__init__(log)
+        self._decrypter = Decrypter(settings.RAS_SEFT_PUBLIC_KEY,
+                                    settings.SDX_SEFT_PRIVATE_KEY,
+                                    settings.SDX_SEFT_PRIVATE_KEY_PASSWORD)
+
+        self.quarantine_publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_QUARANTINE_QUEUE)
+
+        self._ftp = SDXFTP(logger, settings.FTP_HOST, settings.FTP_USER, settings.FTP_PASS, settings.FTP_PORT)
+
+    def on_message(self, unused_channel, basic_deliver, properties, body):
+
+        delivery_count = self.get_delivery_count_from_properties(properties)
+
+        tx_id = self.get_tx_id_from_properties(properties)
+        if not tx_id:
+            self.quarantine_publisher.publish_message(body)
+            self.reject_message(basic_deliver.delivery_tag)
+            logger.error("No tx_id so quarantining message",
+                         action="quarantined",
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+
+        encrypted_jwt = body.decode("utf-8")
+        logger.debug("Message Received", tx_id=tx_id)
+
+        try:
+            decrypted_payload = self._decrypter.decrypt(encrypted_jwt)
+        except DecryptError as e:
+            # Move to the quarantine queue
+            self.quarantine_publisher.publish_message(body)
+            self.reject_message(basic_deliver.delivery_tag, tx_id=tx_id)
+            logger.error("Bad decrypt",
+                         action="quarantined",
+                         exception=e,
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+        except Exception as e:
+            self.nack_message(basic_deliver.delivery_tag, tx_id=tx_id)
+            logger.error("Failed to process",
+                         action="nack",
+                         exception=e,
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+
+        file_contents = decrypted_payload.get("file")
+        file_name = decrypted_payload.get("filename")
+        logger.debug("Decrypted file", file_name=file_name, tx_id=tx_id)
+
+        decoded_contents = base64.b64decode(file_contents)
+
+        try:
+            self._ftp.deliver_binary(settings.FTP_FOLDER, file_name, decoded_contents)
+            logger.debug("Delivered to FTP server", tx_id=tx_id)
+            self.acknowledge_message(basic_deliver.delivery_tag)
+        except IOError as e:
+            logger.exception("Unable to deliver to the FTP server", e)
+            self.nack_message(basic_deliver.delivery_tag, tx_id=tx_id)
+
+    @staticmethod
+    def get_tx_id_from_properties(properties):
+        """
+        Returns the tx_id for a message from a rabbit queue. The value is
+        auto-set by rabbitmq.
+        """
+        tx_id = None
+        try:
+            if properties.headers:
+                tx_id = properties.headers['tx_id']
```

Comment:
It'll raise a KeyError and get caught in the except, logging that it's missing a tx_id